### DISCUSSION TITLE
dropdown doesn't show selected item

### DISCUSSION THREAD

**Discussion by tgarijo** [primefaces/community#1843] [PrimeNG]
Hi, I have a issue that cannot resolve. 
I fill a dropdown with this structure

Angular 17.0.6
Primeng 17.12.0
 ```
{
        "id": 1,
        "name": "Bed & Breakfast",
        "abbreviation": "BB",
        "active": false,
        "deleted": false,
        "observations": null,
        "created": "lunes, 01 de enero de 0001 0:00",
        "createdBy": "tgarijo",
        "lastModify": "",
        "lastModifiedBy": null
    },
    {
        "id": 2,
        "name": "Half Board",
        "abbreviation": "HB",
        "active": false,
        "deleted": false,
        "observations": null,
        "created": "lunes, 01 de enero de 0001 0:00",
        "createdBy": "tgarijo",
        "lastModify": "",
        "lastModifiedBy": null
    }
```
My html is

```
 <td class="meanPlan">
            <p-dropdown
              #MealPlanSelect
              [options]="MealPlan"
              optionLabel="abbreviation"
              formControlName="MealPlan"
              dataKey="MealPlan" <!--  if I remove doesn't  work either -->
              >
            </p-dropdown>        
  </td>
```
And my typescript is (of course I omitted http get)

```
public Hotels : Hotel[] = []; // Hotels to display
public MealPlan : MealPlan[] = []; // MealPlan to display


this.hotelFormGroup = this.fb.group ({

      MealPlan: ['', [Validators.required]],
    });
.....

 // hotel from http get 
  this.hotelFormGroup.patchValue({MealPlan: hotel.mealPlan})

```

As you see,  it set the value with patch.  On this specific case the abbreviation is HB, but the dropdown show always "BB", You  can see on the screenshot

![imagen](https://github.com/primefaces/community/assets/7958523/416942aa-a053-4b9b-b0cd-31fce21207c2)

![imagen](https://github.com/primefaces/community/assets/7958523/335b5a80-5801-401c-bcf6-520174375bfb)

It  will hope setting  to "HB" and show this value on dropdown .

Anyone help me?

kinds regards

**Comment by 3P1L-0** [comment]
I'm also facing this same problem and still found no official solution. The closest I have come to an official statement would be [this issue's marked answer](https://github.com/orgs/primefaces/discussions/819#discussioncomment-8010871)
